cogs/afterhours/afterhours.py

```diff
@@ -0,0 +1,126 @@
+        """
+        ctxGuild = channel.guild
+        ctxChannelId = await self.config.guild(ctxGuild).channelId()
+        ctxChannel = discord.utils.get(ctxGuild.channels, id=ctxChannelId)
+        if not ctxChannel:
+            self.logger.error("Cannot find channel to construct context!")
+            return None
+        async for message in ctxChannel.history(limit=1):
+            lastMessage = message
+        return await self.bot.get_context(lastMessage)
+
+    async def makeStarboardChanges(
+        self, ctx: Context, channel: discord.abc.GuildChannel, remove=False
+    ):
+        self.logger.info("Applying/removing Starboard exceptions, remove=%s", remove)
+        sbCog = self.bot.get_cog("Starboard")
+        if not sbCog:
+            self.logger.error("Starboard not loaded. skipping")
+            return
+
+        try:
+            starboard = sbCog.starboards[ctx.guild.id]["highlights"]
+        except KeyError:
+            self.logger.error("Cannot get the starboard!")
+
+        if remove:
+            await ctx.invoke(sbCog.blacklist_remove, starboard=starboard, channel_or_role=channel)
+        else:
+            await ctx.invoke(sbCog.blacklist_add, starboard=starboard, channel_or_role=channel)
+
+    async def notifyChannel(self, ctx, remove=False):
+        if remove:
+            await ctx.send(f":information_source: **{AH_CHANNEL} removed, removing exceptions**")
+        else:
+            await ctx.send(f":information_source: **{AH_CHANNEL} created, adding exceptions**")
+
+    @commands.Cog.listener("on_guild_channel_create")
+    async def handleChannelCreate(self, channel: discord.abc.GuildChannel):
+        self.logger.info(
+            "Channel creation has been detected. Name: %s, ID: %s", channel.name, channel.id
+        )
+
+        if channel.name == AH_CHANNEL:
+            self.logger.info("%s detected, applying exceptions", AH_CHANNEL)
+            ctx = await self.getContext(channel)
+            if not ctx:
+                return
+            await self.notifyChannel(ctx)
+            await self.makeStarboardChanges(ctx, channel)
+            await self.config.guild(channel.guild).afterHoursChannelId.set(channel.id)
+
+    @commands.Cog.listener("on_guild_channel_delete")
+    async def handleChannelDelete(self, channel: discord.abc.GuildChannel):
+        self.logger.info(
+            "Channel deletion has been detected. Name: %s, ID: %s", channel.name, channel.id
+        )
+        channelId = await self.config.guild(channel.guild).afterHoursChannelId()
+        if channelId and channel.id == channelId:
+            self.logger.info("%s detected, removing exceptions", AH_CHANNEL)
+            ctx = await self.getContext(channel)
+            if not ctx:
+                return
+            await self.notifyChannel(ctx, remove=True)
+            await self.makeStarboardChanges(ctx, channel, remove=True)
+            await self.config.guild(channel.guild).afterHoursChannelId.set(None)
```

Comment:
Cannot assume only one channel. Recall archiving channels, they may persist for 1-3 days.